### PR TITLE
Replace Chart mock data used in Experiment Reports with the endpoind

### LINKED ISSUE #24530: [UI] Replace mock data used in Experiment Reports with the endpoind
### Parent Issue

#23625

### User Story

As a user, I want the mock data to be replaced with the data generated by the backend API in the DotExperimentsReportsChartComponent component so that I can have accurate information about experiments.

### Acceptance Criteria

- The DotExperimentsReportsChartComponent component display data generated by the backend API.
- The data displayed in the frontend matches the data in the backend.

### Proposed Objective

Core Features

### Proposed Priority

Priority 2 - Important

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.

_No response_

### Assumptions & Initiation Needs

- The backend API is functional and provides accurate data.
- The frontend application is able to make requests to the backend API.
- The frontend application is able to parse and display the data received from the backend API.
- The DotExperimentsReportsChartComponent  component are currently using mock data.

### Quality Assurance Notes & Workarounds

_No response_

### Sub-Tasks & Estimates
- [x] Is Empty state of Line Chart

### PR DESCRIPTION
**Pull request by oidacra**
### Proposed Changes
* Replace Chart mock data used in Experiment Reports with the endpoind

### Checklist
- [x] Tests
- [X] Translations
- [ ] Security Implications Contemplated (add notes if applicable)

### Additional Info
N/A

### Screenshots
ChartJS Graphic
Screenshot 2023-04-13 at 4 19 35 PM

Empty State
Screenshot 2023-04-14 at 11 22 44 AM

### <samp>🤖 Generated by Copilot at ec4121c</samp>

Refactored and improved the experiments reports portlet, especially the chart component and its data handling. Renamed and updated the message keys and the model types for the portlet. Added loading, empty and data states for the chart, using Angular and PrimeNG modules. Increased the test coverage and readability of the spec files.

## Related Issue
Fixes #24530 

# Explanation of Changes

### <samp>🤖 Generated by Copilot at ec4121c</samp>

*  Rename `getDotExperimentChartJsOptions` to `getDotExperimentLineChartJsOptions` for clarity ([link](https://github.com/dotCMS/core/pull/24628/files?diff=unified&w=0#diff-551a8e81c825f75bbada7637c279b41b36f3ca538d1f934d04215e502f86d371L8-R8))
*  Comment out ticks callback to prevent label overlap on x-axis ([link](https://github.com/dotCMS/core/pull/24628/files?diff=unified&w=0#diff-551a8e81c825f75bbada7637c279b41b36f3ca538d1f934d04215e502f86d371L67-R69))
*  Add `ngIf` and `ng-template` to show loading, empty or data states of the chart ([link](https://github.com/dotCMS/core/pull/24628/files?diff=unified&w=0#diff-3d4f5269471e2cd6412c7e1bfbf9f7109d4d0e02e21bc092b6c3303d6569798fL1-R43))
*  Add styles for the empty state using background image and text colors ([link](https://github.com/dotCMS/core/pull/24628/files?diff=unified&w=0#diff-178387d7c89f1a21c441de7fe18fb324e3ca5011464565e6d850ac6a73ae05ecR9-R31))
*  Import `CHARTJS_DATA_MOCK` from test mocks file ([link](https://github.com/dotCMS/core/pull/24628/files?diff=unified&w=0#diff-e65867a18f39fa166da660ea22d3ce9fa4885003c621c0ec0436d14719cffa35R7))
*  Use `fdescribe` to focus on test suite ([link](https://github.com/dotCMS/core/pull/24628/files?diff=unified&w=0#diff-e65867a18f39fa166da660ea22d3ce9fa4885003c621c0ec0436d14719cffa35L12-R17))
*  Add tests for loading and empty states and use `setInput` method ([link](https://github.com/dotCMS/core/pull/24628/files?diff=unified&w=0#diff-e65867a18f39fa166da660ea22d3ce9fa4885003c621c0ec0436d14719cffa35L36-R66))
*  Import `NgIf`, `OnChanges`, `SimpleChanges`, `Input`, `SkeletonModule` and `SpinnerModule` from Angular and PrimeNG modules ([link](https://github.com/dotCMS/core/pull/24628/files?diff=unified&w=0#diff-3da1b23139d5886b8aaed2918f23a09083104c40c93a3d94f0ba36aed9a63dc9L3-R11))
*  Add `SkeletonModule` and `SpinnerModule` to component imports array ([link](https://github.com/dotCMS/core/pull/24628/files?diff=unified&w=0#diff-3da1b23139d5886b8aaed2918f23a09083104c40c93a3d94f0ba36aed9a63dc9L20-R53))
*  Remove chart skeleton content from `DotExperimentsReportsSkeletonComponent` template ([link](https://github.com/dotCMS/core/pull/24628/files?diff=unified&w=0#diff-502aba42f63a4f6072450c43dad5b61b6ff854efbc8eab250d445e881701d52fL39-L43))
*  Pass `chartConfig`, `data` and `loading` inputs to `DotExperimentsReportsChartComponent` ([link](https://github.com/dotCMS/core/pull/24628/files?diff=unified&w=0#diff-4b655b7b588e91ad4508341799411fc1d4c2fe2c3db84f43cbf7729a8f35a395L29-R41))
*  Use `experiments` prefix instead of `dot.experimental` for message keys ([link](https://github.com/dotCMS/core/pull/24628/files?diff=unified&w=0#diff-4b655b7b588e91ad4508341799411fc1d4c2fe2c3db84f43cbf7729a8f35a395L43-R59), [link](https://github.com/dotCMS/core/pull/24628/files?diff=unified&w=0#diff-4b655b7b588e91ad4508341799411fc1d4c2fe2c3db84f43cbf7729a8f35a395L124-R128))
*  Add variable for max container width of chart and table ([link](https://github.com/dotCMS/core/pull/24628/files?diff=unified&w=0#diff-98e4b242386163f8ba65d07757e0304c0b74f4b51992062547040c054be0c733R3-R4))
*  Use variable for max container width and increase container padding ([link](https://github.com/dotCMS/core/pull/24628/files?diff=unified&w=0#diff-98e4b242386163f8ba65d07757e0304c0b74f4b51992062547040c054be0c733L26-R32), [link](https://github.com/dotCMS/core/pull/24628/files?diff=unified&w=0#diff-98e4b242386163f8ba65d07757e0304c0b74f4b51992062547040c054be0c733L56-R57))
*  Import `DotExperimentsReportsChartComponent` from its file ([link](https://github.com/dotCMS/core/pull/24628/files?diff=unified&w=0#diff-308ab8a75fd20b72419ef2b637ca1fea7aee92df526895685122b99eca6ca5c4R18))
*  Use `chartData` instead of `chartResults` in `defaultVmMock` object ([link](https://github.com/dotCMS/core/pull/24628/files?diff=unified&w=0#diff-308ab8a75fd20b72419ef2b637ca1fea7aee92df526895685122b99eca6ca5c4L48-R49))
*  Add test for `DotExperimentsReportsChartComponent` existence when not loading ([link](https://github.com/dotCMS/core/pull/24628/files?diff=unified&w=0#diff-308ab8a75fd20b72419ef2b637ca1fea7aee92df526895685122b99eca6ca5c4R118-R124))
*  Import `RippleModule` from PrimeNG module ([link](https://github.com/dotCMS/core/pull/24628/files?diff=unified&w=0#diff-ed2fea3550be4fe34af13b12e3ced9145f1df039b57d5c89f5dacb6fdc0868d0R8))
*  Add `RippleModule` to component imports array ([link](https://github.com/dotCMS/core/pull/24628/files?diff=unified&w=0#diff-ed2fea3550be4fe34af13b12e3ced9145f1df039b57d5c89f5dacb6fdc0868d0L48-R50))
*  Add `chartConfig` property to pass to `DotExperimentsReportsChartComponent` ([link](https://github.com/dotCMS/core/pull/24628/files?diff=unified&w=0#diff-ed2fea3550be4fe34af13b12e3ced9145f1df039b57d5c89f5dacb6fdc0868d0L57-R63))
*  Add `defaultVariantId` property and use it in template ([link](https://github.com/dotCMS/core/pull/24628/files?diff=unified&w=0#diff-ed2fea3550be4fe34af13b12e3ced9145f1df039b57d5c89f5dacb6fdc0868d0L96-R103))
*  Import `DotMessageService` and `MockDotMessageService` from data access and utils testing modules ([link](https://github.com/dotCMS/core/pull/24628/files?diff=unified&w=0#diff-7a41974e260813c23b2e75a570d8ef3b34b4bb6ec10bf9653819a0e063d54cc5L7-R9))
*  Add `messageServiceMock` object with mock messages for weekdays ([link](https://github.com/dotCMS/core/pull/24628/files?diff=unified&w=0#diff-7a41974e260813c23b2e75a570d8ef3b34b4bb6ec10bf9653819a0e063d54cc5R34-R43))
*  Add `messageServiceMock` object to test bed providers array ([link](https://github.com/dotCMS/core/pull/24628/files?diff=unified&w=0#diff-7a41974e260813c23b2e75a570d8ef3b34b4bb6ec10bf9653819a0e063d54cc5R57-R60))
*  Remove `chartResults` property from `expectedState` object ([link](https://github.com/dotCMS/core/pull/24628/files?diff=unified&w=0#diff-7a41974e260813c23b2e75a570d8ef3b34b4bb6ec10bf9653819a0e063d54cc5L68-R84))
*  Add `getChartData$` observable to `combineLatest` operator ([link](https://github.com/dotCMS/core/pull/24628/files?diff=unified&w=0#diff-7a41974e260813c23b2e75a570d8ef3b34b4bb6ec10bf9653819a0e063d54cc5R156-R224))
*  Import `ChartData` type from chart.js module ([link](https://github.com/dotCMS/core/pull/24628/files?diff=unified&w=0#diff-2776f530268a3c582f32b83baacac6909948c3ac1a9b0824a483b0d66c5cbc00R2), [link](https://github.com/dotCMS/core/pull/24628/files?diff=unified&w=0#diff-928c7a4710d662672ace21bcd6d05d9c1af34829cfb3f0bc7c7df00c552b3a5aR1))
*  Import `DotMessageService` from data access module ([link](https://github.com/dotCMS/core/pull/24628/files?diff=unified&w=0#diff-2776f530268a3c582f32b83baacac6909948c3ac1a9b0824a483b0d66c5cbc00L10-R26))
*  Import `daysOfTheWeek` and `DEFAULT_VARIANT_ID` constants from dotcms models module ([link](https://github.com/dotCMS/core/pull/24628/files?diff=unified&w=0#diff-2776f530268a3c582f32b83baacac6909948c3ac1a9b0824a483b0d66c5cbc00L10-R26), [link](https://github.com/dotCMS/core/pull/24628/files?diff=unified&w=0#diff-928c7a4710d662672ace21bcd6d05d9c1af34829cfb3f0bc7c7df00c552b3a5aR12))
*  Remove `chartResults` property from `DotExperimentsReportsState` interface and `initialState` object ([link](https://github.com/dotCMS/core/pull/24628/files?diff=unified&w=0#diff-2776f530268a3c582f32b83baacac6909948c3ac1a9b0824a483b0d66c5cbc00L26), [link](https://github.com/dotCMS/core/pull/24628/files?diff=unified&w=0#diff-2776f530268a3c582f32b83baacac6909948c3ac1a9b0824a483b0d66c5cbc00L33-R42))
*  Add `getChartData$` observable to transform experiment and results data into chart data object ([link](https://github.com/dotCMS/core/pull/24628/files?diff=unified&w=0#diff-2776f530268a3c582f32b83baacac6909948c3ac1a9b0824a483b0d66c5cbc00R77-R85))
*  Add `dotMessageService` parameter to constructor and inject it as dependency ([link](https://github.com/dotCMS/core/pull/24628/files?diff=unified&w=0#diff-2776f530268a3c582f32b83baacac6909948c3ac1a9b0824a483b0d66c5cbc00L119-R137))
*  Add `chartData` property to view model object and use `getChartData$` observable to get its value ([link](https://github.com/dotCMS/core/pull/24628/files?diff=unified&w=0#diff-2776f530268a3c582f32b83baacac6909948c3ac1a9b0824a483b0d66c5cbc00L130-R144))
*  Add helper methods to generate and format chart data from experiment and results data ([link](https://github.com/dotCMS/core/pull/24628/files?diff=unified&w=0#diff-2776f530268a3c582f32b83baacac6909948c3ac1a9b0824a483b0d66c5cbc00R151))
*  Change `ExperimentResultsMocks` object to use `DEFAULT_VARIANT_ID` constant and add more details data ([link](https://github.com/dotCMS/core/pull/24628/files?diff=unified&w=0#diff-928c7a4710d662672ace21bcd6d05d9c1af34829cfb3f0bc7c7df00c552b3a5aL131-R150))
*  Add `CHARTJS_DATA_MOCK` object with mock data for line chart component ([link](https://github.com/dotCMS/core/pull/24628/files?diff=unified&w=0#diff-928c7a4710d662672ace21bcd6d05d9c1af34829cfb3f0bc7c7df00c552b3a5aR158-R179))
*  Change `variants` property of `DotResultGoal` interface to use union type of `'DEFAULT' | string` ([link](https://github.com/dotCMS/core/pull/24628/files?diff=unified&w=0#diff-253959797c88a1b66365ca0759703481a484a03c59dc027e2dfc6accc6cf2b77L33-R33))
*  Change `DefaultExperimentChartDatasetColors` object to an array of `LineChartColorsProperties` type and rename it to `ExperimentChartDatasetColorsVariants` ([link](https://github.com/dotCMS/core/pull/24628/files?diff=unified&w=0#diff-253959797c88a1b66365ca0759703481a484a03c59dc027e2dfc6accc6cf2b77L174-R197))
*  Change message keys for experiments portlet to use `experiments` prefix and add messages for chart title and empty state ([link](https://github.com/dotCMS/core/pull/24628/files?diff=unified&w=0#diff-21db9723fa4c52389731b7575f1e19bd364aa992b7e549e22c08a33456994e94L5231-R5247))

### CODE REVIEW

File: core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-reports/dot-experiments-reports.component.html [hunk 1]
```diff
@@ -26,33 +26,37 @@
...
+                                {{ 'experiments.reports.clicks' | dm }}
                             </div>
                             <div class="flex col flex-wrap align-items-center justify-content-end">
-                                {{ 'dot.experimental.reports.best-variant' | dm }}
+                                {{ 'experiments.reports.best-variant' | dm }}
                             </div>
                             <div class="flex col flex-wrap align-items-center justify-content-end">
-                                {{ 'dot.experimental.reports.improvement' | dm }}
+                                {{ 'experiments.reports.improvement' | dm }}
                             </div>
```
Comment (zJaaal) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-reports/dot-experiments-reports.component.html:60]: What about an ngFor here?

File: core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-reports/store/dot-experiments-reports-store.ts
Comment (hmoreras) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-reports/store/dot-experiments-reports-store.ts:1]: maybe is worth it to move data parse logic for the chart and details to a service.

**Review by hmoreras — APPROVED** [review]

File: core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-reports/dot-experiments-reports.component.html (hunk 1 above)
Comment (oidacra) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-reports/dot-experiments-reports.component.html:60]: Yes, could be a NgFor. I don't know if @hmoreras takes that into consideration in the issue about connect details with real data (https://github.com/dotCMS/core/issues/24626).

File: core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-reports/store/dot-experiments-reports-store.ts
Comment (oidacra) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-reports/store/dot-experiments-reports-store.ts:1]: I will move it; less code in the store.

**Review by zJaaal — APPROVED** [review]

File: core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-reports/components/dot-experiments-reports-chart/dot-experiments-reports-chart.component.scss [hunk 2]
```diff
@@ -6,3 +6,26 @@
...
...='0.922339'/%3E%3Cline x1='61.9375' y1='395.461' x2='31.5003' y2='395.461' stroke='%23F4F4F6' stroke-width='0.922339'/%3E%3Cline x1='61.9375' y1='437.461' x2='31.5003' y2='437.461' stroke='%23F4F4F6' stroke-width='0.922339'/%3E%3Crect x='0.5' y='10' width='31' height='16' rx='8' fill='%23F4F4F6'/%3E%3Crect x='0.5' y='52' width='31' height='16' rx='8' fill='%23F4F4F6'/%3E%3Crect x='0.5' y='93' width='31' height='16' rx='8' fill='%23F4F4F6'/%3E%3Crect x='0.5' y='136' width='31' height='16' rx='8' fill='%23F4F4F6'/%3E%3Crect x='0.5' y='180' width='31' height='16' rx='8' fill='%23F4F4F6'/%3E%3Crect x='0.5' y='220' width='31' height='16' rx='8' fill='%23F4F4F6'/%3E%3Crect x='0.5' y='261' width='31' height='16' rx='8' fill='%23F4F4F6'/%3E%3Crect x='0.5' y='304' width='31' height='16' rx='8' fill='%23F4F4F6'/%3E%3Crect x='0.5' y='346' width='31' height='16' rx='8' fill='%23F4F4F6'/%3E%3Crect x='0.5' y='386' width='31' height='16' rx='8' fill='%23F4F4F6'/%3E%3Crect x='0.5' y='430' width='31' height='16' rx='8' fill='%23F4F4F6'/%3E%3Cdefs%3E%3CclipPath id='clip0_3061_22368'%3E%3Crect width='855' height='515' fill='white' transform='translate(61.5)'/%3E%3C/clipPath%3E%3C/defs%3E%3C/svg%3E");
```
Comment (rjvelazco) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-reports/components/dot-experiments-reports-chart/dot-experiments-reports-chart.component.scss:11]: Is there a `material icon` for this?

**Review by rjvelazco — APPROVED** [review]

File: core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-reports/components/dot-experiments-reports-chart/dot-experiments-reports-chart.component.scss (hunk 2 above)
Comment (zJaaal) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-reports/components/dot-experiments-reports-chart/dot-experiments-reports-chart.component.scss:11]: PrimeNG Icons would be better, I think that material icons will be replaced by it

Comment (oidacra) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-reports/components/dot-experiments-reports-chart/dot-experiments-reports-chart.component.scss:11]: > Is there a `material icon` for this?

No is this image:

Screenshot 2023-04-14 at 3 48 15 PM

Comment (oidacra) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-reports/components/dot-experiments-reports-chart/dot-experiments-reports-chart.component.scss:11]: > 

I know. It is a background image.
